components/PromptRestore.vue

```diff
@@ -187,11 +193,11 @@ export default {
       const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
       const timeFormat = escapeHtml( this.$store.getters['prefs/get'](TIME_FORMAT));
 
-      const name = snapshot.id.split(':');
-      const d = day(snapshot.created).format(dateFormat);
-      const t = day(snapshot.created).format(timeFormat);
+      const created = snapshot.createdAt || snapshot.created || snapshot.metadata.creationTimestamp;
+      const d = day(created).format(dateFormat);
+      const t = day(created).format(timeFormat);
 
-      return `${ d } ${ t } : ${ name[1] }`;
+      return `${ d } ${ t } : ${ snapshot.name }`;
```

Comment:
For RKE2 snapshots in the snapshot list on the cluster provisioning detail page we use ``snapshotFile.name` for the name, so there's a discrepancy here with the snapshot.name.
```
on-demand-rk2-pool1-e453ad1d-d85zn-1647019230 
```
vs
```
rk2-on-demand-rk2-pool1-e453ad1d-d85zn-1647019230-local
```

For RKE1 snapshots we use the normal `nameDisplay`